Src/Support/Google.Apis.Auth/GoogleJsonWebSignature.cs

```diff
@@ -14,15 +14,191 @@
 limitations under the License.
 */
 
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Json;
+using Google.Apis.Util;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
 namespace Google.Apis.Auth
 {
     /// <summary>
     /// Google JSON Web Signature as specified in https://developers.google.com/accounts/docs/OAuth2ServiceAccount.
     /// </summary>
     public class GoogleJsonWebSignature
     {
-        // TODO(peleyal): We should provide a way to verify JWS.
-        // Take a look at: https://github.com/googleplus/gplus-verifytoken-csharp/blob/master/verifytoken.ashx.cs.
+        /// <summary>
+        /// An exception that is thrown when a Json Web Token (JWT) is invalid.
+        /// </summary>
+        public class InvalidJwtException : Exception
```

Comment:
Nit, not fond of inner class used for exception. Is this a pattern in the rest of the code?